module/LearnZF2Themes/src/LearnZF2Themes/Factory/ReloadConfigFactory.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/**
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license.
+ */
+
+namespace LearnZF2Themes\Factory;
+
+use Zend\ServiceManager\ServiceLocatorInterface;
+
+/**
+ * @author Stanimir Dimitrov <stanimirdim92@gmail.com>
+ */
+final class ReloadConfigFactory
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function __invoke(ServiceLocatorInterface $serviceLocator)
+    {
+        $request = $serviceLocator->get('Request');
+
+        if ($request->isPost()) {
+            $config = $serviceLocator->get('Config');
+            $themeName = $request->getPost()['themeName'];
+            $serviceLocator->setAllowOverride(true);
+            $config['theme']['name'] = $themeName;
+            $serviceLocator->setService('Config', $config);
+            $serviceLocator->setAllowOverride(false);
+        }
+
+        return $serviceLocator;
```

Comment:
imo, the zf2 factory should create another instance that may pull service from serviceLocator, not a serviceLocator instance itself, @acelaya do you have some opinion about it ? I think it can be replaced by SharedEventManager, in Module class: 

```php
use LearnZF2Themes\Service\ReloadService;

$shm->attach(ReloadService::class, 'reload',  function($e) use ($this->service) {
      $reloadService = $e->getTarget();
      
      // do the re-set config here..., or use attachAggregate with passed listener \
      // that handle it  
      $reloadService->getEventManager()
                              ->attachAggregate($this->service->get('ConfigChangerListener'));
});
```

in `ConfigChangerListener`, we can do: 

and in the `LearnZF2Themes\Service\ReloadService` , we can do : 

```php

namespace LearnZF2Themes\Service;

use Zend\EventManager\EventManager;
use Zend\EventManager\EventManagerAwareInterface;
use Zend\EventManager\EventManagerInterface;
use Zend\EventManager\EventManagerAwareTrait;
 
class ReloadService implements EventManagerAwareInterface 
{
   use EventManagerAwareTrait;
   
   public function reload() 
   {
        $this->getEventManager()->trigger(__FUNCTION__, $this);
    }
}
```

So, in controller, we can do: 
```php
$this->reloadService->reload();
```

more clear imo ^^